invenio_sip2/records/record.py

```diff
@@ -0,0 +1,222 @@
+# -*- coding: utf-8 -*-
+#
+# INVENIO-SIP2
+# Copyright (C) 2020 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""API for manipulating the client."""
+
+from copy import deepcopy
+from uuid import uuid4
+
+from invenio_sip2 import datastore
+
+from ..errors import ServerAlreadyRunning
+
+
+class Sip2RecordMetadata(dict):
+    """Sip2RecordMetadata class."""
+
+    record_type = None
+
+    def __init__(self, data, **kwargs):
+        """Initialize instance with dictionary data.
+
+        :param data: Dict with record metadata.
+        """
+        super(Sip2RecordMetadata, self).__init__(data or {})
+
+    @classmethod
+    def create(cls, data, id_=None, **kwargs):
+        """Create record.
+
+        :param data: Dict with metadata.
+        :param id_: Specify a UUID to use for the new record.
+        """
+        assert cls.record_type
+        # TODO: check if record already exist and raise exception
+        if not id_:
+            id_ = str(uuid4())
```

Comment:
python tricks : you can use `id_ = id_ or str(uuid4())` if you want

Reply:
thanks